### DISCUSSION TITLE
RealTime database Api is not working

### DISCUSSION THREAD

**Discussion by khare4** [appwrite/appwrite#5418] [General]
I have tried to delete, update, and create  new document in my database but nothing is happening in console its just printing  subscription: http://192.168.50.78/realtime?project=64386b13977ba52c89d08&channels%5B%5D=documents, when i make changes to my database nothing shows in console, here is my code how i have implemented, 

```dart
Databases db = Databases(client);
final realtime = Realtime(client);
RealtimeSubscription? subscription;
var items;

loadItems() async {
  try {
    final res = await db.listDocuments(
      databaseId: databaseId,
      collectionId: collectionId,
    );
    setState(() {
      print(res.documents.map((e) => e.$id));
      items = res.documents;
    });
  } on AppwriteException catch (e) {
    print(e.message);
  }
}

void subscribe() {
  try {
    final realtime = Realtime(client);
    subscription = realtime.subscribe(['documents']);
    subscription!.stream.listen((data) {
      if (data.payload.isNotEmpty) {
        switch (data.events) {
          case ["database.documents.create"]:
            var item = data.payload;
            items.add(item);
            setState(() {});
            break;
          case ["database.documents.delete"]:
            var item = data.payload;
            items.removeWhere((it) => it['\$id'] == item['\$id']);
            setState(() {});
            break;
          default:
            break;
        }
      }
    });
  } on AppwriteException catch (e) {
    print(e.message);
  }
}
```

**Comment by loks0n** [comment]
Hi there 👋

Thanks for providing your code. It seems like the issue lies in the event names you're trying to subscribe to in your code versus the actual event names provided by Appwrite's Realtime API.

In your current code, you're listening for "database.documents.create" and "database.documents.delete" events, but based on the Appwrite API, the correct event names should be "databases.\*.collections.\*.documents.\*.create" and "databases.\*.collections.\*.documents.\*.delete".

These events are specific not only to the database but also to a particular collection and the document in that collection. The * is a wildcard that means any database, any collection, and any document.

Here is how you can adjust your code:
```dart
void subscribe() {
  try {
    final realtime = Realtime(client);
    subscription = realtime.subscribe(['databases.*.collections.*.documents.*']);
    subscription!.stream.listen((data) {
      if (data.payload.isNotEmpty) {
        var item = data.payload;
        if (data.events.contains("databases.*.collections.*.documents.*.create")) {
            items.add(item);
            setState(() {});
        } else if (data.events.contains("databases.*.collections.*.documents.*.delete")) {
            items.removeWhere((it) => it['\$id'] == item['\$id']);
            setState(() {});
        }
      }
    });
  } on AppwriteException catch (e) {
    print(e.message);
  }
}
```

In this snippet, the subscription listens to all document events in all collections of all databases. We then use the contains() function to check if the specific event we're interested in is included in the list of events (data.events). You can replace the wildcards (*) with the actual IDs if you want to listen to specific databases, collections, or documents.

I hope this helps! Feel free to reach out if you encounter any more issues 💬 
Happy coding 🚀
